### DISCUSSION TITLE
Why doesn't useRouter's asPath work properly in the next / head Head tag?

### DISCUSSION THREAD

**Discussion by okamoto-tosei** [vercel/next.js#36878] [Help]
Why can't I get a query even if I use asPath to get the path in the Head tag?
I don't understand why asPath doesn't work properly with this code.

```
const  {asPath} =useRouter();

<Head>
   <meta property="og:url" content={http://localhost:8080 + asPath} />
</Head>
```

What is being output
http://localhost:8080/foo/[code]

**Comment by icyJoseph** [comment]
Hi,

We cannot really see your localhost deploy, but I did this on a page:

```tsx
import type { NextPage } from "next";
import Head from "next/head";
import { useRouter } from "next/router";

const Home: NextPage = () => {
  const { asPath } = useRouter();

  return (
    <>
      <Head>
        <meta property="og:url" content={asPath}></meta>
      </Head>

      <div>Hello World</div>
    </>
  );
};

export default Home;
```

And both in dev and production mode it outputs:

![Screenshot 2022-05-13 at 08 13 40](https://user-images.githubusercontent.com/21013447/168222217-b4f0b68c-7b04-429d-9b76-c415f118fdd8.png)


Perhaps you have conflicting head tags that apply the same property? or something like that? Assuming your code was pseudocode because it has syntax errors.

**Reply by okamoto-tosei** [reply]
@icyJoseph 

thank you for your answer!

It was because there were multiple conflicts in the code and I was updating the execution of the code ...

Thanks to you, I was able to solve it! Thank you!
